src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -87,9 +87,10 @@ public SpoonFolder getParent() {
 	@Override
 	public List<SpoonFolder> getSubFolders() {
 		List<SpoonFolder> result = new ArrayList<>();
+		Set<SpoonFolder> distinctSet = new HashSet<>();
 		for (SpoonFile f : getAllFiles()) {
 			SpoonFolder folder = f.getParent();
-			if (!result.contains(folder)) {

```

Comment:
are your sure hashCode/equals of SpoonFolder are correct? I'm not sure they are well tested.